### PR TITLE
Tools Provisioner

### PR DESCRIPTION
**Pull request by tomjn**
Implements #2270 

## Checks

* [x] I've updated the changelog.
* [x] I've tested this PR
* [x] This PR is for the `develop` branch not the `stable` branch.
* [x] This PR is complete and ready for review.

### CODE REVIEW

**Comment by tomjn** [comment]
This is running ok for a VM that's already provisioned and updated to this code.

It needs testing for a brand new VVV too confirm it does not break provisioning

File: provision/provision.sh [hunk 1]
```diff
@@ -57,19 +57,17 @@ export VVV_PACKAGE_REMOVAL_LIST=()
 . "/srv/provision/core/nginx/provision.sh"
 . "/srv/provision/core/memcached/provision.sh"
 . "/srv/provision/core/php/provision.sh"
-. "/srv/provision/core/composer/provision.sh"
-. "/srv/provision/core/nodejs/provision.sh"
-. "/srv/provision/core/grunt/provision.sh"
 . "/srv/provision/core/mailhog/provision.sh"
-. "/srv/provision/core/wp-cli/provision.sh"
-. "/srv/provision/core/phpcs/provision.sh"
+. "/srv/provision/core/nodejs/provision.sh"
```
Comment (Mte90) [inline provision/provision.sh:61]: this isn't a tool after all? should be part of that provisioner?

File: provision/provision.sh [hunk 2]
```diff
@@ -57,19 +57,17 @@ export VVV_PACKAGE_REMOVAL_LIST=()
...
 
 ### SCRIPT
 #set -xv
 
 vvv_hook init
 
+# If you need to disable this check then something is terribly wrong, tell us on github/slack
```
Comment (Mte90) [inline provision/provision.sh:68]: maybe is better to add a parameter in `config.yml` so people don't need to hack with the VVV code?

**Review by Mte90 — CHANGES_REQUESTED** [review]

File: provision/provision.sh (hunk 1 above)
Comment (tomjn) [reply provision/provision.sh:61]: Which? PHPCS and WP CLI are tools so they're moved to the tools provisioner, Node is a base dependency

File: provision/provision.sh (hunk 2 above)
Comment (tomjn) [reply provision/provision.sh:68]: We've discussed this elsewhere and the conclusion was "tell us how it failed and we can fix it", if the feature si broken we should fix it, adding an option called `bypass_broken_check: true` is just silly

File: provision/provision.sh (hunk 1 above)
Comment (Mte90) [reply provision/provision.sh:61]: it was node but ok

Comment (tomjn) [reply provision/provision.sh:61]: Ah, Node isn't a tool

**Comment by Mte90** [comment]
```
    default:  ▷ Running the 'tools' provisioner...
    default:  * Running tools_install
    default:  ▷ Running tools_setup hook
    default:  * Turning off XDebug to avoid Composer performance issues
    default:  ✘ Disabling active PHP v8.0 debug mod: 'xhgui'
    default:  ✘ Disabling active PHP v7.4 debug mod: 'xhgui'
    default:  ✘ Disabling active PHP v7.3 debug mod: 'xhgui'
    default:  ✘ Disabling active PHP v8.0 debug mod: 'tideways_xhprof'
    default:  ✘ Disabling active PHP v7.4 debug mod: 'tideways_xhprof'
    default:  ✘ Disabling active PHP v7.3 debug mod: 'tideways_xhprof'
    default:  * Restarting PHP FPM services so that the change takes effect
    default:  ✔ All PHP Debug mods are now turned off.
    default:  * Checking if Composer is installed
    default:  * Making sure the Composer cache is not owned by root
    default:  * Checking for GitHub tokens
    default:  * Checking for Composer updates
    default:  ▷ Running after_composer hook
    default:  * Provisioning PHP_CodeSniffer (phpcs), see https://github.com/squizlabs/PHP_CodeSniffer
    default: Loading composer repositories with package information
    default: Updating dependencies
    default: Nothing to modify in lock file
    default: Installing dependencies from lock file (including require-dev)
    default: Nothing to install, update or remove
    default: Generating autoload files
    default:  * Setting WordPress-Core as the default PHPCodesniffer standard
    default:  * The following PHPCS standards are set up:
    default:  * PHPCS provisioning has ended
    default:  ✔ Finished after_composer hook in 3s
    default:  * Updating Grunt CLI
    default:  * Updated Grunt CLI
    default:  * Installing/updating WP-CLI
    default:  * Updating wp-cli...
    default:  * WP CLI Nightly updated
    default:  * Disabling debug mods if present before running wp package installs
    default:  * Restarting PHP FPM services so that the change takes effect
    default:  ✔ All PHP Debug mods are now turned off.
    default:  * Updating WP packages
    default:  * WP package updates completed
    default:  ✔ Finished tools_setup hook in 16s
    default:  * Finalizing Tools
    default:  ▷ Running tools_finalize hook
    default:  ✔ The 'tools' provisioner completed in 16 seconds.
```
Seems working with no troubles

**Review by Mte90 — APPROVED** [review]

**Pull request merged**